cpp/src/arrow/ipc/reader.cc

```diff
@@ -899,13 +898,23 @@ Status ReadSparseCSRIndex(const flatbuf::SparseTensor* sparse_tensor,
   ARROW_ASSIGN_OR_RAISE(auto indices_data,
                         file->ReadAt(indices_buffer->offset(), indices_buffer->length()));
 
-  std::vector<int64_t> indptr_shape({shape[0] + 1});
   std::vector<int64_t> indices_shape({non_zero_length});
-
-  *out = std::make_shared<SparseCSRIndex>(
-      std::make_shared<Tensor>(indptr_type, indptr_data, indptr_shape),
-      std::make_shared<Tensor>(indices_type, indices_data, indices_shape));
-  return Status::OK();
+  switch (sparse_index->compressedAxis()) {
+    case flatbuf::SparseMatrixCompressedAxis_Row: {
+      std::vector<int64_t> indptr_shape({shape[0] + 1});
```

Comment:
Perhaps we should check that `shape` has the right size here? Or is it done elsewhere already?

Reply:
It's checked when calling the [constructor](https://github.com/apache/arrow/blob/8f4a212e1278dc32e6d9ca2e9db3e602f8971473/cpp/src/arrow/sparse_tensor.h#L229) with [ValidateSparseCSXIndex](https://github.com/apache/arrow/blob/8f4a212e1278dc32e6d9ca2e9db3e602f8971473/cpp/src/arrow/sparse_tensor.cc#L726)

Reply:
Exactly. We should confirm that the length of `shape` is 2, here.  I'll add it.

@rok `ValidateSparseCSXIndex` doesn't check the shape of a matrix.